sdk/table/azure-table/azure/table/_shared/table_client_base.py

```diff
@@ -0,0 +1,60 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse  # type: ignore
+
+from azure.table._shared._error import _validate_table_name
+from azure.table._shared.base_client import parse_query
+from .base_client import StorageAccountHostsMixin
+
+class TableClientBase(StorageAccountHostsMixin):
+    def __init__(
+        self, account_url, # type: str
+        table_name, # type: str
+        credential=None, # type: Union[str,TokenCredential]
+        **kwargs # type: Any
+    ):
+        # type: (...) -> None
+        """Create TableClientBase from a Credential.
+
+        :param account_url:
+            A url to an Azure Storage account.
+        :type account_url: str
+        :param table_name: The table name.
+        :type table_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+        :type credential: Union[str,TokenCredential]
+
+        :returns: None
+        """
+
+        _validate_table_name(table_name)
+
+        try:
+            if not account_url.lower().startswith('http'):
+                account_url = "https://" + account_url
+        except AttributeError:
+            raise ValueError("Account URL must be a string.")
+        parsed_url = urlparse(account_url.rstrip('/'))
+        if not table_name:
+            raise ValueError("Please specify a table name.")
+        if not parsed_url.netloc:
+            raise ValueError("Invalid URL: {}".format(parsed_url))
+
+        _, sas_token = parse_query(parsed_url.query)
```

Comment:
Would it be possible to also add the other ways of initializing a client to this base, i.e. `from_connection_string`

Reply:
I'm not sure about this one. In the lilaw_async doesn't have any of the other ways of initializing and I assume you would be able to, I'm not 100% sure if those methods would have to be asynced. @kristapratico probably has better insight than I do

Reply:
`from_connection_string` is an authentication method exposed to the user, it shouldn't be in the base class